### PR TITLE
Pr/more clipboard improvements

### PR DESCRIPTION
**Pull request by uli42**

### CODE REVIEW

**Comment by sunweaver** [comment]
Please rebase pr/more_clipboard_improvements, thanks. The rebase is not trivial due to the pr/xlib_types merge.

**Comment by uli42** [comment]
done

File: nx-X11/programs/Xserver/hw/nxagent/Clipboard.c [hunk 1]
```diff
@@ -1579,10 +1579,8 @@ void nxagentSetSelectionOwner(Selection *pSelection)
     {
       #ifdef DEBUG
       fprintf(stderr, "%s: lastSelectionOwner.client [%p] index [%d] -> [%p] index [%d]\n", __func__,
-              (void *)lastSelectionOwner[i].client,
-              CLINDEX(lastSelectionOwner[i].client),
-              (void *)pSelection->client,
-              CLINDEX(pSelection->client));
+              (void *)lastSelectionOwner[i].client, CLINDEX(lastSelectionOwner[i].client),
+              (void *)pSelection->client, CLINDEX(pSelection->client));
```
Comment (sunweaver) [inline nx-X11/programs/Xserver/hw/nxagent/Clipboard.c]: commit msg does not match what is happening in this commit

**Review by sunweaver — APPROVED** [review]
Wow. Quite a bit of work, this one. LGTM

**Pull request merged**